### ISSUE TITLE
.go scale

### ISSUE THREAD

**Issue by Lunarria** [EpsilonWoW/Enteleaie#57]
When I came back online after a brief brake I had some of my objects scaled back to their original size, but I had as well few objects preserving their new scale. For an example, I had several WallTiles scaled and like 50% of them preserved their scale, and the rest retracted to their original scale.

**Comment by AddieA** [comment]
Did you use a macro to scale them? Did you by any chance add 1 at the end of the command? Are all the objects the same type?

**Comment by Lunarria** [comment]
I didn't use macros, no. And yes, I did, but I might have not used 1 on some of the objects since I'm used to RPH and objects saves on there regardless without or with the 1. And, lastly, yes, these were all tiles.

**Comment by timgblack** [comment]
Is this reproducible?

**Comment by timgblack** [comment]
If this is not reproducible, I can't fix it. I'm going to say that this can't be replicated reliably.

If it can, please re-open this ticket with more information.

**Comment by timgblack** [comment]
Apparently this can be recreated but I don't know how.

Find out or I can't fix it - I don't have the time to investigate why it happens.

**Comment by Aarontravess** [comment]
I also have no idea how to recreate the problem, it's completely random.

It's as simple as this.

1. I spawn objects.
2. I scale them.
3. Some of the scales remain, some don't. There is no pattern, infact some of the objects have been spawned one after the other, and even then one remains scaled and one resets to 1.

Could be something to do with the building filter, or the object limit applied to accounts. I really don't know. I spawned a load of objects today and none of them kept their scale.

**Comment by timgblack** [comment]
According to Skype, this was resolved afterwards. I think it was the case that the scale of new objects worked fine until a relog (at which point they reset) and after a crash (after which they looked fine permanently).

A potential change to the way objects work for #28 could see a solution for this too, but I haven't started on said changes.

**Comment by Aarontravess** [comment]
This has now completely baffled my mind. Now there are random objects that will reset after a relog, but they now don't even fix after a restart. Most of these objects were scaled below 1, for example the UL_Power_orb02 I've been using is massive at scale 1, and I have to scale them to 0.01 to get them to the scale I want. 

Now it's almost impossible to build just because they've all returned to scale 1 and engulfed most of my phase.

**Comment by AddieA** [comment]
I want to suggest a little test. We make it mandatory so one has to save an object's scale so it remains in the database after a crash, hopefully. We see how objects behave then and maybe we can easier see what causes the issue. I we are unable to fix it, we can make object scale save permanent.

**Comment by timgblack** [comment]
Deciding whether to save scales or not just means objects will not retain their scales after a crash if they aren't saved, and they might if they are. I don't really see what benefit this would have towards figuring out the problem.

I'll see if I can't change things around a bit to get better handling of setting scales on creation.

I still need GUIDs and scales (what it was set to, what the core thinks and what you see if it's different from what the core reports).

**Comment by AddieA** [comment]
Those were spawned in my phase, ID 28. Owner is 14. GUIDS:

330272
330273
330271

I spawned them, scaled them up to 2.5. Exited the game. .go info returned the right scale. The server did not restart in the meantime. Returned after 20 minutes, the scales reverted back to 1.

Those were spawned in Main Phase. Owner is 14. GUIDS:

330268
330269
330270

I spawned them, scaled them up to 2.5. Exited the game. .go info returned the right scale. The server did not restart in the meantime. Returned after 20 minutes, the scales reverted back to 1.
